### PR TITLE
Replace Mappers with Foreachers in several passes

### PR DESCRIPTION
**Pull request by albert-magyar**
Drops ~40 LoC and helps with clarity in some cases by eliminating spurious return value expressions.

### CODE REVIEW

File: src/main/scala/firrtl/passes/CheckInitialization.scala [hunk 1]
```diff
@@ -85,10 +81,10 @@ object CheckInitialization extends Pass {
       }
     }
 
-    c.modules foreach {
+    c.modules.foreach({
```
Comment (chick) [inline src/main/scala/firrtl/passes/CheckInitialization.scala]: Nit: Parens seem extraneous here

**Comment by jackkoenig** [comment]
As an added bonus, this modestly improves performance, about 3-5%. This improvement most likely comes from avoiding unnecessary object creations as done by the mappers.

**Comment by albert-magyar** [comment]
> As an added bonus, this modestly improves performance, about 3-5%. This improvement most likely comes from avoiding unnecessary object creations as done by the mappers.

So what you're saying is that I can write an ISCA paper about this.

File: src/main/scala/firrtl/passes/CheckInitialization.scala (hunk 1 above)
Comment (albert-magyar) [reply src/main/scala/firrtl/passes/CheckInitialization.scala]: True! I removed them.

**Pull request merged**